lib/ansible/modules/cloud/vmware/vmware_guest.py

```diff
@@ -1319,6 +1346,9 @@ def deploy_vm(self):
         self.configure_network(vm_obj=vm_obj)
         self.configure_cdrom(vm_obj=vm_obj)
 
+        # set the destination datastore for VM & disks
+        (datastore, datastore_name) = self.select_datastore(vm_obj, destfolder, self.configspec)
```

Comment:
Any specific reason to move this here ?

Reply:
select_datastore cluster feature need destfolder and configspec that are set in previous steps